Forward-port downstream pcie-brcmstb patches to 6.14

P33M opened the pull request:
This applies on top of the upstream bcm2712-aware pcie-brcmstb driver and restores downstream bugfixes and compatibility features.

The intent is to keep all the conflated 2712/platform-specific adjustments relatively self-contained, hopefully avoiding merge conflicts until they land upstream.

Commit:
dt-bindings: interrupt-controller: Add bcm2712 MSI-X DT bindings

Adds DT bindings for bcm2712 MSI-X interrupt peripheral controller.

Signed-off-by: Stanimir Varbanov <svarbanov@suse.de>

Commit:
PCI: brcmstb: Add bcm2712 support

Add bare minimum amount of changes in order to support PCIe RC hardware
IP found on RPi5. The PCIe controller on bcm2712 is based on bcm7712 and
as such it inherits register offsets, perst, bridge_reset ops and inbound
windows count.
Although, the implementation for bcm2712 needs a workaround related to the
control of the bridge_reset where turning off of the root port must not
shutdown the bridge_reset and this must be avoided. To implement this
workaround a quirks field is introduced in pcie_cfg_data struct.

Signed-off-by: Stanimir Varbanov <svarbanov@suse.de>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>

P33M committed:
PCI: brcmstb: don't use ASPM state defines for register bits

In commit b478e162f227 ("PCI/ASPM: Consolidate link state defines")
PCIE_LINK_STATE_L1 and PCIE_LINK_STATE_L0s grew some bits for more
granular control of ASPM.

This broke the aspm-no-l0s override, instead disabling link ASPM
completely if this DT property was specified.

Specify the field bits in the driver.

Fixes: caab002d5069 ("PCI: brcmstb: Disable L0s component of ASPM if requested")
Fixes: 0693b4207fd7 ("PCI: brcmstb: Split post-link up initialization to brcm_pcie_start_link()")
Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>

Commit:
irqchip: Add Broadcom bcm2712 MSI-X interrupt controller

Add an interrupt controller driver for MSI-X Interrupt Peripheral (MIP)
hardware block found in bcm2712. The interrupt controller is used to
handle MSI-X interrupts from peripherials behind PCIe endpoints like
RP1 south bridge found in RPi5.

There are two MIPs on bcm2712, the first has 64 consecutive SPIs
assigned to 64 output vectors, and the second has 17 SPIs, but only
8 of them are consecutive starting at the 8th output vector.

Signed-off-by: Stanimir Varbanov <svarbanov@suse.de>

Commit:
PCI: brcmstb: Reuse config structure

Instead of copying fields from pcie_cfg_data structure to
brcm_pcie reference it directly.

Signed-off-by: Stanimir Varbanov <svarbanov@suse.de>
Reviewed-by: Florian Fainelil <florian.fainelli@broadcom.com>

Commit:
PCI: brcmstb: Adjust PHY PLL setup to use a 54MHz input refclk

The default input reference clock for the PHY PLL is 100Mhz, except for
some devices where it is 54Mhz like bcm2712C1 and bcm2712D0.

To implement this adjustments introduce a new .post_setup op in
pcie_cfg_data and call it at the end of brcm_pcie_setup function.

The bcm2712 .post_setup callback implements the required MDIO writes that
switch the PLL refclk and also change PHY PM clock period.

Without this RPi5 PCIex1 is unable to enumerate endpoint devices on
the expansion connector.

Signed-off-by: Stanimir Varbanov <svarbanov@suse.de>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>

Commit:
PCI: brcmstb: Fix for missing of_node_put

A call to of_parse_phandle() increments refcount, of_node_put must be
called when done the work on it. Fix missing of_node_put() on the
msi_np device node by using scope based of_node_put() cleanups.

Cc: stable@vger.kernel.org # v5.10+
Fixes: 40ca1bf580ef ("PCI: brcmstb: Add MSI support")
Signed-off-by: Stanimir Varbanov <svarbanov@suse.de>

Commit:
arm64: dts: broadcom: bcm2712: Add PCIe DT nodes

Add PCIe devicetree nodes, plus needed reset and mip MSI-X
controllers.

Signed-off-by: Stanimir Varbanov <svarbanov@suse.de>

Commit:
arm64: dts: broadcom: bcm2712-rpi-5-b: Enable PCIe DT nodes

Enable pcie1 and pcie2 DT nodes. Pcie1 is used for the extension
connector and pcie2 is used for RP1 south-bridge.

Signed-off-by: Stanimir Varbanov <svarbanov@suse.de>

P33M committed:
PCI: brcmstb: set BCM7712/2712-specific AXI bridge handling behaviours

These chips use a UBUS-AXI bridge component that has configurable
timeout and error response handling.

Suppress AXI error responses to CPU requests, otherwise these are fatal
if they reach the ARM cluster, and set reasonably large timeouts for
both Mem and Cfg requests.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>

P33M committed:
PCI: brcmstb: add support for BCM2712 priority forwarding

The BCM2712 root complexes can interpret priority signalling in two
different ways, based on the incoming Traffic Class of a TLP.

The TLP TCs are assigned to separate internal request/response queues,
and assigned different AXI IDs. These queues can have outgoing AXI
transactions tagged based on:

- Static QoS values
- Dynamic QoS through internal backpressure
- Dynamic QoS with elevation based on Vendor Messages received by the RC

The VDM mechanism is of limited use due to implementation bugs, but the
implicit reordering due to separate ID assignment allows higher-priority
traffic from an EP to overtake other traffic in the RC and rest of the
system.

RP1 assigns TCs based on its internal bus managers, and internally tags
read requests to allow out-of-order completions, so these two features
operate in concert to provide priority service to e.g. MIPI camera or
display traffic.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>

P33M committed:
dt-bindings: pci: pcie-brcmstb: add BCM2712-specific properties

There is configurable priority forwarding hardware in this variant of the
Root Complex controller. Add optional properties to configure FIFO
backpressure or Vendor-Defined Message priority forwarding.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>

P33M committed:
PCI: brcmstb: Enable CRS software visibility after linkup

It appears that bits in the Root Control Register are reset with
perst_n, which means the PCI layer's call to enable CRS prior to
adding/scanning the bus has no effect. Open-code the enable in
brcm_pcie_start_link as a workaround.

Without CRS visibility, configuration reads issued by the CPU don't
retire if the endpoint returns a CRS response - the RC will poll until a
(large) timeout is reached. This means the core can stall for a long
time during boot.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>

P33M committed:
dt-bindings: pci: pcie-brcmstb: add optional brcm,tperst-clk-ms property

Some platforms may require an extended time with refclk active before
PERST# is released. Add a property to let the RC driver know how long to
wait.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>

P33M committed:
PCI: pcie-brcmstb: optionally extend Tperst_clk time

Some endpoints need longer than the minimum Tperst_clk time of 100us
that the PCIe specification allows for, as they may need to sequence
internal resets off the stable output of internal PLLs prior to removal
of fundamental reset. PCIe switches are an especially bad case, in some
cases requiring up to 100 milliseconds for stable downstream link
behaviour.

Parse the DT property brcm,tperst-clk-ms and use this to hold PERST# low
during brcm_pcie_start_link().

The BRCM RC typically outputs 200us of stable refclk before deasserting
PERST#. By masking/forcing the output signal while deasserting the
internal reset, the effect is to extend the length of time that the
refclk is active and stable before PERST# is released.

The TX lanes will enter the Polling state before PERST# is released, but
this appears to be harmless.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>

jamesequinlan committed:
PCI: brcmstb: Set gen limitation before link, not after

When the user elects to limit the PCIe generation via the appropriate DT
property, apply the settings before the PCIe link-up, not after.

Fixes: c0452137034bda8f686dd9a2e167949bfffd6776 ("PCI: brcmstb: Add Broadcom STB PCIe host controller driver")

Signed-off-by: Jim Quinlan <james.quinlan@broadcom.com>
Fixes: c0452137034b ("PCI: brcmstb: Add Broadcom STB PCIe host controller driver")
Reviewed-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>

jamesequinlan committed:
PCI: brcmstb: Write to internal register to change link cap

The driver was mistakenly writing to a RO config-space register
(PCI_EXP_LNKCAP).  Although harmless in this case, the proper destination
is an internal RW register that is reflected by PCI_EXP_LNKCAP.

Fixes: c0452137034bda8f686dd9a2e167949bfffd6776 ("PCI: brcmstb: Add Broadcom STB PCIe host controller driver")

Signed-off-by: Jim Quinlan <james.quinlan@broadcom.com>
Reviewed-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>

jamesequinlan committed:
PCI: brcmstb: Do not assume that reg field starts at LSB

When setting a register field it was assumed that the field started at the
lsb of the register.  Although the masks do indeed start at the lsb, and
this will probably not change, it is prudent to use a method that makes no
assumption about the mask's placement in the register.

The uXXp_replace_bits() calls are used since they are already prevalent
in this driver.

Signed-off-by: Jim Quinlan <james.quinlan@broadcom.com>
Reviewed-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>

jamesequinlan committed:
PCI: brcmstb: Fix error path upon call of regulator_bulk_get()

If regulator_bulk_get() returns an error, no regulators are created and we
need to set their number to zero.  If we do not do this and the PCIe
link-up fails, regulator_bulk_free() will be invoked and effect a panic.

Also print out the error value, as we cannot return an error upwards as
Linux will WARN on an error from add_bus().

Fixes: 9e6be018b263 ("PCI: brcmstb: Enable child bus device regulators from DT")
Signed-off-by: Jim Quinlan <james.quinlan@broadcom.com>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>

jamesequinlan committed:
PCI: brcmstb: Fix potential premature regulator disabling

Our system for enabling and disabling regulators is designed to work only
on the port driver below the root complex.  The conditions to discriminate
for this case should be the same when we are adding or removing the bus.
Without this change the regulators may be disabled prematurely when a bus
further down the tree is removed.

Fixes: 9e6be018b263 ("PCI: brcmstb: Enable child bus device regulators from DT")
Signed-off-by: Jim Quinlan <james.quinlan@broadcom.com>
Reviewed-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>

jamesequinlan committed:
PCI: brcmstb: Use same constant table for config space access

The constants EXT_CFG_DATA and EXT_CFG_INDEX vary by SOC. One of the
map_bus methods used these constants, the other used different constants.
Fortunately there was no problem because the SoCs that used the latter
map_bus method all had the same register constants.

Remove the redundant constants and adjust the code to use them.  In
addition, update EXT_CFG_DATA to use the 4k-page based config space access
system, which is what the second map_bus method was already using.

Signed-off-by: Jim Quinlan <james.quinlan@broadcom.com>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>

jamesequinlan committed:
PCI: brcmstb: Make two changes in MDIO register fields

The HW team has decided to "tighten" some field definitions in the MDIO
packet format.  Fortunately these two changes may be made in a backwards
compatible manner.

The CMD field used to be 12 bits and now is one.  This change is backwards
compatible because the field's starting bit position is unchanged and the
only commands we've used have values 0 and 1.

The PORT field's width has been changed from four to five bits.  When
written, the new bit is not contiguous with the other four.  Fortunately,
this change is backwards compatible because we have never used anything
other than 0 for the port field's value.

Signed-off-by: Jim Quinlan <james.quinlan@broadcom.com>
Reviewed-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>

jamesequinlan committed:
PCI: brcmstb: Clarify conversion of irq_domain_set_info() param

Just make it clear to the reader that there is a conversion happening, in
this case from an int type to an irq_hw_number_t, an unsigned long int.

Signed-off-by: Jim Quinlan <james.quinlan@broadcom.com>
Reviewed-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>

popcornmix commented:
This booted for me, with working ssd/m2hat and ethernet.

6by9 commented on Documentation/devicetree/bindings/pci/brcm,stb-pcie.yaml:
```diff
@@ -101,6 +101,12 @@ properties:
+      the time between internal release of fundamental reset and
+      the deassertion of the external PERST# pin. This has the
+      effect of increasing the Tperst_clk phase of link init.
+
```
Describe the units and range, eg
```
    $ref: /schemas/types.yaml#/definitions/uint32
    maximum: 5000
```

P33M replied:
Strangely, the bindings checker complains if I define this. But not for uint8 (?)

6by9 commented on Documentation/devicetree/bindings/pci/brcm,stb-pcie.yaml:
```diff
@@ -84,6 +84,11 @@ properties:
+  brcm,fifo-qos-map:
+    description: 2712 only. Each nibble assigns all per-TC FIFOs an AXI
+      priority based on fullness (backpressure signalling).
+      Mutually exclusive with vdm-qos-map.
```
Units and range.

Do we have to encode it within DT, or could it be an array of N values?

Is it mandatory to have one or other of these, or is neither acceptable?

The bit you'll really love is that commenting "2712 only" doesn't wash in bindings. YAML has lots of lovely syntax to denote which compatibles support which properties.
eg https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/i2c/i2c-mux-pca954x.yaml#L95-L126

`make dt_binding_check` checks whether your bindings are valid.
`make CHECK_DTBS=y broadcom/bcm2712-rpi-5-b.dts` checks whether your dtb file complies with the requirements of the binding.
(Add the relevant environment vars if cross-compiling).

P33M replied:
This is where it gets fun: downstream uses nibbles. Would the upstream preference be a u32 array? Bit wasteful to encode 4 bits in 32.

Also, the useful combinations of these properties are either nothing, or "brcm,fifo-qos-map" or "brcm,vdm-qos-map". Can optional properties be specified as mutually exclusive?

6by9 replied:
I would have expected upstream to want an array of u8 or u32, but the only way of finding out would be to ask them.
Wasting <1kB isn't really worth worrying about.

You can convert the DT config directly back into the relevant nibbles for the hardware during probe, so it only throws a small amount extra on the stack during probe.

Yes I believe you can encode mutually exclusion on optional properties. Don't ask me for the exact runes at the moment though.

P33M replied:
I went with an array of u8 - breaks DT compatibility across versions, but them's the breaks.

6by9 commented on drivers/pci/controller/pcie-brcmstb.c:
```diff
@@ -1762,7 +1803,7 @@ static const struct pcie_cfg_data bcm2712_cfg = {
 	.bridge_sw_init_set = brcm_pcie_bridge_sw_init_set_generic,
 	.post_setup	= brcm_pcie_post_setup_bcm2712,
-	.quirks		= CFG_QUIRK_AVOID_BRIDGE_SHUTDOWN,
+	.quirks		= CFG_QUIRK_AVOID_BRIDGE_SHUTDOWN | CFG_QUIRK_NO_SSC,
```
Handling of this SSC quirk should be a separate patch to the UBUS-AXI changes. The commit text doesn't reference this at all.

P33M replied:
Oops - chunk got intermingled during rebase.

6by9 commented on Documentation/devicetree/bindings/pci/brcm,stb-pcie.yaml:
```diff
@@ -96,6 +101,12 @@ properties:
+    description: 2712 only. Each nibble assigns each per-TC FIFO a base AXI
+      priority with elevation depending on Vendor Messages from the EP -
+      specifically, RP1.
+      Mutually exclusive with fifo-qos-map.
```
Ditto

6by9 commented on drivers/pci/controller/pcie-brcmstb.c:
```diff
@@ -1320,10 +1320,11 @@ static int brcm_pcie_setup(struct brcm_pcie *pcie)
-	/* Don't advertise L0s capability if 'aspm-no-l0s' */
-	aspm_support = PCIE_LINK_STATE_L1;
+	/* Always advertise L1 capability */
+	aspm_support = BIT(1);
```
Nicer to define the bits up around the define of register `PCIE_RC_CFG_PRIV1_LINK_CAPABILITY`, and then use them here.

6by9 commented on drivers/pci/controller/pcie-brcmstb.c:
```diff
@@ -1473,13 +1475,32 @@ static int brcm_pcie_start_link(struct brcm_pcie *pcie)
+		writel(tmp, pcie->base + HARD_DEBUG(pcie));
+
+		ret = pcie->cfg->perst_set(pcie, 0);
+		msleep(pcie->tperst_clk_ms);
```
What is the typical value for your sleep?
There were issues for short sleeps with msleep (<30ms), but I believe those are now resolved.

`fsleep(usecs)` is the recommended call - https://elixir.bootlin.com/linux/v6.13.3/source/Documentation/timers/delay_sleep_functions.rst (was Documentation/timers/timers-howto.rst)

P33M replied:
The worst-case EP delay (also the reason for implementing it) is 100ms and that is a minimum. I was unaware of fsleep.

6by9 reviewed (commented):
Seeing as much of this should be destined for upstream, I'm being picky.

And welcome to the lovely world of dt-bindings.

Commit:
dt-bindings: PCI: brcmstb: Update bindings for PCIe on BCM2712

Update PCIe controller bindings with BCM2712 support.

Signed-off-by: Stanimir Varbanov <svarbanov@suse.de>
Acked-by: Rob Herring (Arm) <robh@kernel.org>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>
Tested-by: Ivan T. Ivanov <iivanov@suse.de>
Link: https://lore.kernel.org/r/20250120130119.671119-3-svarbanov@suse.de
[kwilczynski: commit log]
Signed-off-by: Krzysztof Wilczyński <kwilczynski@kernel.org>

Commit:
PCI: brcmstb: Adding a softdep to MIP MSI-X driver

In case brcmstb PCIe driver and MIP MSI-X interrupt controller
drivers are built as modules there could be a race in probing.

To avoid this add a softdep to MIP driver to guarantee that MIP
driver will be load first.

Signed-off-by: Stanimir Varbanov <svarbanov@suse.de>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>
Tested-by: Ivan T. Ivanov <iivanov@suse.de>
Link: https://lore.kernel.org/r/20250120130119.671119-9-svarbanov@suse.de
[kwilczynski: commit log]
Signed-off-by: Krzysztof Wilczyński <kwilczynski@kernel.org>

Commit:
PCI: brcmstb: Expand inbound window size up to 64GB

The BCM2712 memory map can support up to 64GB of system memory, thus
expand the inbound window size in calculation helper function.

The change is safe for the currently supported SoCs that have smaller
inbound window sizes.

Signed-off-by: Stanimir Varbanov <svarbanov@suse.de>
Reviewed-by: Florian Fainelli <florian.fainelli@broadcom.com>
Reviewed-by: Jim Quinlan <james.quinlan@broadcom.com>
Tested-by: Ivan T. Ivanov <iivanov@suse.de>
Link: https://lore.kernel.org/r/20250120130119.671119-6-svarbanov@suse.de
[kwilczynski: commit log]
Signed-off-by: Krzysztof Wilczyński <kwilczynski@kernel.org>

P33M committed:
PCI: brcmstb: add NO_SSC quirk for BCM2712

The PHY MDIO register map is different on BCM2712, and as the PHY input
clock is 54MHz not 100MHz, enabling refclk SSC is both broken and
unfixable.

Mask out attempts to enable SSC with a controller quirk.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>

P33M committed:
DT: bcm2712: swap PCIe QoS properties to the new array type

pcie1 should use the FIFO threshold property as the RC should not
pay attention to Vendor Messages from incompatible endpoint hardware.

Also drop the downstream MPS property, it's no longer needed.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>

P33M commented:
https://lore.kernel.org/linux-arm-kernel/20250214173944.47506-2-james.quinlan@broadcom.com/ obsoletes commit 5ae7f04df6cb5a5223a0d670f3ce7f20b3e43b26 but that's a problem for future us.

6by9 commented:
> https://lore.kernel.org/linux-arm-kernel/20250214173944.47506-2-james.quinlan@broadcom.com/ obsoletes commit [5ae7f04](https://github.com/raspberrypi/linux/commit/5ae7f04df6cb5a5223a0d670f3ce7f20b3e43b26) but that's a problem for future us.

Ish.
These appear to have been accepted, so will be in 6.15/6.16. Whilst we're working on this already, it'd be better to backport those to 6.14 first and fix the problems once, rather than having to revisit it in 3 months time.

It also means that the your patches should apply upstream for when we try submitting them.

(Quickest way to apply the series to the current branch is `b4 shazam 20250214173944.47506-1-james.quinlan@broadcom.com/` - see [b4 docs](https://b4.docs.kernel.org/en/latest/contributor/prep.html#working-with-series-dependencies))

P33M committed:
Revert "PCI: brcmstb: Cast an int variable to an irq_hw_number_t"

This reverts commit bcdeae70d769770532a796f84c0d1a31cb205095.

P33M committed:
Revert "PCI: brcmstb: Make two changes in MDIO register fields"

This reverts commit 97ad2e0917338d8481ac38821b90b9e09f3e2549.

P33M committed:
Revert "PCI: brcmstb: Use same constant table for config space access"

This reverts commit a70c281ae75426e0bacbfa553f4c09965a12c6e2.

P33M committed:
Revert "PCI: brcmstb: Fix potential premature regluator disabling"

This reverts commit 1974bc746d51e8e1b26026b40830ea261e70d7af.

P33M committed:
Revert "PCI: brcmstb: Fix error path upon call of regulator_bulk_get()"

This reverts commit b371b544fafe8e145c7712ce8e1d6dd0b6b45ad0.

P33M committed:
Revert "PCI: brcmstb: Refactor max speed limit functionality"

This reverts commit 5ae7f04df6cb5a5223a0d670f3ce7f20b3e43b26.

P33M committed:
Revert "arm64: dts: broadcom: bcm2712: Add PCIe DT nodes"

This reverts commit 44e8524d7243d59a4fc43773da93763a0b8805de.

P33M committed:
Revert "arm64: dts: broadcom: bcm2712-rpi-5-b: Enable PCIe DT nodes"

This reverts commit 00183d277798cecac273d2cfa717ff8b0001932e.

P33M committed:
Revert "PCI: brcmstb: Fix for missing of_node_put"

This reverts commit 0422c8f319932a7b464a63d7ae27fd176fea5e4a.

P33M committed:
Revert "PCI: brcmstb: Adding a softdep to MIP MSI-X driver"

This reverts commit 13f04a21268ad5ba5cff8710e64c22989ac2bc77.

P33M committed:
Revert "PCI: brcmstb: Adjust PHY PLL setup to use a 54MHz input refclk"

This reverts commit 6467161da991f598d8b763e8d292223f844555bc.

P33M committed:
Revert "PCI: brcmstb: Add bcm2712 support"

This reverts commit abf2169e8ea52c61d5c7ff5e62745d0bcc137d6c.

P33M committed:
Revert "PCI: brcmstb: Expand inbound window size up to 64GB"

This reverts commit 622a3ff428334c976d362a587c3ce75dbfb5c726.

P33M committed:
Revert "PCI: brcmstb: Reuse config structure"

This reverts commit 634b0d02d60d9247395158559dfe8b90b4ba6ac8.

P33M committed:
Revert "irqchip: Add Broadcom bcm2712 MSI-X interrupt controller"

This reverts commit ccd1f2f27c57e93c76d96fb893226b12b0c6b8a0.

P33M committed:
Revert "dt-bindings: PCI: brcmstb: Update bindings for PCIe on bcm2712"

This reverts commit db0549cf8607041fc2f89d535c9e8898bce63e49.

P33M committed:
Revert "dt-bindings: interrupt-controller: Add bcm2712 MSI-X DT bindings"

This reverts commit c4d0832cd6f4d82f0bfafbe3c21e586bb2dfb85b.

P33M commented:
This now 
- reverts v1/v5 of the upstream patch set
- applies v2/v6 of the same
- puts downstream fixes on top of these

The bindings pass dt_binding_check (at least, I think so - lots of others fail).

popcornmix commented:
This booted for me, with working ssd/m2hat and ethernet.

pelwell commented:
@6by9 Is there any reason I shouldn't hit the Merge button?

6by9 commented:
> @6by9 Is there any reason I shouldn't hit the Merge button?

I don't think there's a good reason not to.

Picking up the newer versions of the upstream patches is fine, and P33M is the expert on what the downstream patches do.
DT bindings will always be a pain when it comes to upstreaming them, but they look reasonable to me.

pelwell commented:
Thanks!
